### DISCUSSION TITLE
Need Help Understanding COMMAND_GO_TO = 4 in crazyflie-lib-python/cflib/crazyflie/high_level_commander.py

### DISCUSSION THREAD

**Discussion by jimdaokeke** [bitcraze/discussions#1590] [Q&A]
Hi everyone,

I'm currently working on a research project involving Cognitive Map Learning (CML) using Bitcraze's Crazyflie drones. We're trying to control the drones in a Webots simulation but are encountering some issues.

While going through the code, specifically the crazyflie-lib-python/cflib/crazyflie/high_level_commander.py file, we came across the line COMMAND_GO_TO = 4. and the following 
```
        if self._cf.platform.get_protocol_version() < 8:
            if linear:
                print('Warning: Linear mode not supported in protocol version < 8, update your crazyflie\'s firmware')
            self._send_packet(struct.pack('<BBBfffff',
                                          self.COMMAND_GO_TO,
                                          group_mask,
                                          relative,
                                          x, y, z,
                                          yaw,
                                          duration_s))
        else:
```
 We've been trying to understand the exact meaning of this command and how does it accomplished, but we've had no luck finding documentation or an explanation of how it works.

Could anyone help explain what COMMAND_GO_TO = 4 is used for and how does done? Does it relate to specific movement commands, and how should we be using it in practice for navigating the drone?

We’d really appreciate any guidance or resources that could shed some light on this.

Thanks in advance!

**Comment by gemenerik** [comment]
We plan using [piecewise 7th order no-jerk polynomials](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/pptraj.c#L413). This is based on a pr from the Crazyswarm team, so [their docstring should also apply](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py#L228-L238). You can also look at [how the command is interpreted on the CF side](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp_commander_high_level.c#L431-L436).

The specific bit of  code you included is there for compatibility reasons. In https://github.com/bitcraze/crazyflie-lib-python/pull/470 and https://github.com/bitcraze/crazyflie-firmware/pull/1410, a new `linear` parameter has been introduced that will make the CF go to the setpoint with constant velocity. This uses the COMMAND_GO_TO_2 packet.

**Reply by gemenerik** [reply]
Opened an issue regarding the lack of documentation https://github.com/bitcraze/crazyflie-lib-python/issues/487
